ethereum/eip712/eip712.go

```diff
@@ -235,8 +235,8 @@ func traverseFields(
 			// then continue as normal
 		}
 
-		// If its a nil pointer, do not include in types
-		if fieldType.Kind() == reflect.Ptr && field.IsNil() {
+		// If field is an empty value, do not include in types, since it will not be present in the object
+		if field.IsZero() {
```

Comment:
what's the difference with `IsNil()`?

Reply:
`IsNil()` does not apply to basic primitives:
https://pkg.go.dev/reflect#Value.IsNil